### DISCUSSION TITLE
GitHub action runners - can't activate docker image within (failed to create shim task: OCI runtime create failed: runc create failed)

### DISCUSSION THREAD

**Discussion by JanneKiiskila** [community/community#45763] [Actions]
### Select Topic Area

Product Feedback

### Body

Hei,

do others have issues with starting Dockers from within the GitHub runners? We have a CI job that has worked earlier, but it has now started failing today. It seems it can't start the Docker image that the job is trying to run.

```
Sending build context to Docker daemon  101.5MB

Step 1/11 : FROM ubuntu:20.04
 ---> 680e5dfb52c7
Step 2/11 : ARG developer_certificate=./config/mbed_cloud_dev_credentials.c
 ---> Using cache
 ---> 7acde406f4c3
Step 3/11 : ARG update_certificate=./config/update_default_resources.c
 ---> Using cache
 ---> eb2a3be10370
Step 4/11 : WORKDIR /usr/src/app/mbed-edge
 ---> Using cache
 ---> c029a1e1dc1c
Step 5/11 : RUN apt-get update &&     DEBIAN_FRONTEND="noninteractive" apt-get -y install tzdata &&     apt-get install -y build-essential libc6-dev cmake python3.6 python3-pip python3-setuptools &&     apt-get install -y vim python3-venv
 ---> Using cache
 ---> 41[17](https://github.com/PelionIoT/mbed-edge/actions/runs/4041694538/jobs/6948606647#step:3:20)2859ccd6
Step 6/11 : COPY . .
 ---> aca6cedd1200
Step 7/11 : RUN pip3 install --upgrade pip
 ---> Running in 2e206fd3b324
failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: error during container init: error setting cgroup config for procHooks process: failed to write "c *:* m": write /sys/fs/cgroup/devices/docker/2e206fd3b32455bd29c[18](https://github.com/PelionIoT/mbed-edge/actions/runs/4041694538/jobs/6948606647#step:3:21)deac2e4d5054f25830fc49f81e848564a83[24](https://github.com/PelionIoT/mbed-edge/actions/runs/4041694538/jobs/6948606647#step:3:27)73cde8/devices.allow: operation not permitted: unknown
Error: Process completed with exit code 1.
```

Seems something is off in the GitHub runner? Status page for GitHub however looks all green.

The job is in a public repo, fails and earlier succeeding runs can be seen there.
https://github.com/PelionIoT/mbed-edge/actions/runs/4041694538/jobs/6948606647

Best Regards,

Janne Kiiskilä

**Comment by jge162** [comment]
Hi @JanneKiiskila the fail message is likely due to a permissions issue on the host system. Try checking if the ->`user running` the Docker container has sufficient `permissions to write to the cgroup devices.`

<img width="1027" alt="Screenshot 2023-01-30 at 12 31 20 AM" src="https://user-images.githubusercontent.com/31228460/215426227-c2a07de6-5cd6-4fff-988e-2485655b74e0.png">

**Comment by JanneKiiskila** [comment]
Editing my answer - it's indeed running in a locally hosted runner instead. Something has now changed in that which causes this issue. We will look into it, thanks for the tip.

**Reply by jge162** [reply]
Sorry I was not able to help. Just going off the error message.

**Reply by jge162** [reply]
You can always get help from [GitHub support](https://support.github.com/request/account?tags=hubberfy_account) by submitting a help ticket. Have a great rest of your week.)

<img width="1034" alt="Screenshot 2023-01-30 at 12 46 45 AM" src="https://user-images.githubusercontent.com/31228460/215432663-7f57bf37-c892-44e3-a9c3-2236ab96efbb.png">
